toolkit/block-producer-metadata/pallet/src/lib.rs

```diff
@@ -131,6 +141,17 @@ pub mod pallet {
 		/// Should return the chain's genesis UTXO
 		fn genesis_utxo() -> UtxoId;
 
+		/// The currency used for holding tokens
+		type Currency: MutateHold<Self::AccountId, Reason = Self::RuntimeHoldReason>
+			+ frame_support::traits::tokens::fungible::Mutate<Self::AccountId>;
+
+		/// The amount of tokens to hold when upserting metadata
+		#[pallet::constant]
+		type HoldAmount: Get<<Self::Currency as frame_support::traits::tokens::fungible::Inspect<Self::AccountId>>::Balance>;
```

Comment:
please move it to separate type
  ```rust
  type BalanceOf<T> = <<Self as Config>::Currency as frame_support::traits::tokens::fungible::Inspect<Self::AccountId>>::Balance>
  ```
  ```suggestion
		type HoldAmount: Get<BalanceOf<Self>>;
```

Reply:
Okay, I see there is such a convention in polkdot-sdk. I 'll try to add BalanceOf<T>. Your example doesn't use `T` and is syntactically incorrect.